vkcs/kubernetes/resource_cluster.go

```diff
@@ -272,23 +291,72 @@ func ResourceKubernetesCluster() *schema.Resource {
 				Computed:    true,
 				Description: "Enables syncing of security policies of cluster. Default value is false.",
 			},
+			"cluster_type": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Default:          clusterTypeStandard,
+				ForceNew:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice([]string{clusterTypeStandard, clusterTypeRegional}, false)),
+				Description:      "Type of the kubernetes cluster, may be `standard` or `regional`. Default type is `standard`.",
+			},
 		},
-		Description: "Provides a kubernetes cluster resource. This can be used to create, modify and delete kubernetes clusters.",
+		CustomizeDiff: resourceKubernetesClusterCustomizeDiff,
+		Description:   "Provides a kubernetes cluster resource. This can be used to create, modify and delete kubernetes clusters.",
 	}
 }
 
-func resourceKubernetesClusterCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	config := meta.(clients.Config)
-	containerInfraClient, err := config.ContainerInfraV1Client(util.GetRegion(d, config))
-	if err != nil {
-		return diag.Errorf("error creating container infra client: %s", err)
+func resourceKubernetesClusterCustomizeDiff(_ context.Context, diff *schema.ResourceDiff, _ any) error {
+	newClusterType := diff.Get("cluster_type").(string)
+	switch newClusterType {
+	case clusterTypeStandard:
+		if masterCount, ok := diff.GetOk("master_count"); ok {
+			mCount := masterCount.(int)
+			if mCount != 1 && mCount != 3 && mCount != 5 {
```

Comment:
Let's add constants for min and max values and check for the value to be in this range and to be odd.

Reply:
maybe use `validation.ToDiagFunc(validation.IntInSlice([]int{1, 3, 5}))` in schema?

Reply:
i check []int{1, 3, 5} in schema and check 3,5 if cluster type is regional in code below

Reply:
Well, it's a bit messy, i think. Double validation, and coupled a bit.